js/content/common.js

```diff
@@ -2120,20 +2120,41 @@ let Downloader = (function(){
 
     let self = {};
 
-    self.download = function(options) {
-        if (!options.url) {
+    self.download = async function(options) {
+        if (options.content) {
             let blob = new Blob([ options.content ], {type : "text/plain;charset=UTF-8"});
             options.url = URL.createObjectURL(blob);
         }
 
-        let element = document.createElement('a');
-        element.setAttribute('href', options.url);
-        element.setAttribute('download', options.filename);
-        element.style.display = 'none';
+        if (options.url && !options.url.startsWith("data:")) {
+            options.url = await self.toDataURL(options.url).catch(console.error);
+        }
+
+        let element = document.createElement("a");
+        element.setAttribute("href", options.url);
+        element.setAttribute("download", options.filename || "download");
+        element.style.display = "none";
         document.body.appendChild(element);
         element.click();
         document.body.removeChild(element);
     };
+
+    self.toDataURL = function(url) {
+        return new Promise(function(resolve, reject) {
+            RequestData.getBlob(url, {credentials: "omit"})
+            .then(function(blob) {
+                let fr = new FileReader();
```

Comment:
Why a `FileReader`?

Isn't replacing this function with `URL.createObjectURL(blob)` a strict improvement?

Why even fetch the blob instead of just creating the anchor and setting the url on it?